src/vocab.py

```diff
@@ -0,0 +1,136 @@
+import os
+import sys
+import pathlib
+import argparse
+import collections
+from collections import OrderedDict, defaultdict
+from typing import List, Tuple, Iterable, Union
+
+from ordered_set import OrderedSet
+from logging_util import ColoredLog
+
+project_root = pathlib.Path(__file__).parents[1]
+
+class Vocabulary(object):
+    """
+    A vocabulary instance, for tokens, intents, and slots.
+    """
+
+    def __init__(self, name: str, speical_tokens: List=None):
+        """Initialization.
+
+        :name: name of the vocabulary
+        :speical_tokens: speical tokens to be added to vocabulary
```

Comment:
Can you clarify more what "special tokens" are and what they are used for?

Reply:
Added more details!